shell/platform/windows/flutter_windows_engine_unittests.cc

```diff
@@ -1111,5 +1113,39 @@ TEST_F(FlutterWindowsEngineTest, LifecycleStateToFrom) {
   }
 }
 
+TEST_F(FlutterWindowsEngineTest, ChannelListenedTo) {
+  FlutterWindowsEngineBuilder builder{GetContext()};
+  builder.SetDartEntrypoint("enableLifecycleToFrom");
+
+  auto window_binding_handler =
+      std::make_unique<::testing::NiceMock<MockWindowBindingHandler>>();
+  MockFlutterWindowsView view(std::move(window_binding_handler));
+  view.SetEngine(builder.Build());
+  FlutterWindowsEngine* engine = view.GetEngine();
+  EngineModifier modifier(engine);
+  modifier.embedder_api().RunsAOTCompiledDartCode = []() { return false; };
+  auto old_run = modifier.embedder_api().Run;
+
+  static bool listened_to = false;
+  modifier.embedder_api().Run = MOCK_ENGINE_PROC(
+      Run, ([old_run](size_t version, const FlutterRendererConfig* config,
+                      const FlutterProjectArgs* args, void* user_data,
+                      FLUTTER_API_SYMBOL(FlutterEngine) * engine_out) {
+        FlutterProjectArgs new_args = *args;
+        new_args.channel_update_callback =
```

Comment:
Correct me if I'm wrong, but this test is effectively the same as `EmbedderTest.RegisterChannelListener`? Both check that the embedder's `channel_update_callback` receives the expected values.

Perhaps a better test for this layer would be that `WindowsLifecycleManager::BeginProcessingLifecycle` is called when the framework adds the `flutter/lifecycle` listener. This could be done by using the `EngineModifier` to set a mocked lifecycle manager.

Please let me know if I'm missing something!